src/api/functions/push-directory.ts

```diff
@@ -1,39 +1,50 @@
-import { FilterPredicate, Options } from "../../types";
+import { FilterPredicate, Options, TransformPredicate } from "../../types";
 
 export type PushDirectoryFunction = (
   directoryPath: string,
   paths: string[],
-  filters?: FilterPredicate[]
+  filters?: FilterPredicate[],
+  transformer?: TransformPredicate
 ) => void;
 
 function pushDirectoryWithRelativePath(root: string): PushDirectoryFunction {
-  return function (directoryPath, paths) {
-    paths.push((directoryPath || ".").substring(root.length));
+  return function (directoryPath, paths, _filters, transformer) {
+    const relativePath = (directoryPath || ".").substring(root.length);
+    paths.push(transformer?.(relativePath, true) ?? relativePath);
```

Comment:
optional chaining would sadly break compatibility with anything above node 14

Reply:
Oh my, got used to those features so much at work that I forgot they weren't supported at one point. Gonna change later.